sklearn/cluster/_agglomerative.py

```diff
@@ -759,6 +759,9 @@ class AgglomerativeClustering(ClusterMixin, BaseEstimator):
     n_connected_components_ : int
         The estimated number of connected components in the graph.
 
+        .. versionchanged:: 0.21
+            ``n_components_`` was renamed to ``n_connected_components_``.
+
```

Comment:
I'm not sure if we need these. Its true that the attribute was renamed, but from the perspective of `n_connected_components_`, nothing has changed since its introduction. Not sure if we should do a versionadded instead or not. WDYT @jnothman ?

Reply:
I feel like these might be useful, e.g. for a  user to understand why and when their code started to break. No strong opinion though

Reply:
Maybe `versionadded` and:

> `n_connected_components_` was added to replace `n_components_`

Reply:
+1 for the @thomasjpfan `versionadded` formulation.